stellarphot/tests/test_core.py

```diff
@@ -25,74 +25,172 @@
 
 
 def test_camera_attributes():
+    # Check that the attributes are set properly
+    data_unit = u.adu
     gain = 2.0 * u.electron / u.adu
     read_noise = 10 * u.electron
     dark_current = 0.01 * u.electron / u.second
     pixel_scale = 0.563 * u.arcsec / u.pix
+    max_val = 50000 * u.adu
+
     c = Camera(
+        data_unit=data_unit,
         gain=gain,
         read_noise=read_noise,
         dark_current=dark_current,
         pixel_scale=pixel_scale,
+        max_data_value=max_val,
     )
+    assert c.data_unit == data_unit
     assert c.gain == gain
     assert c.dark_current == dark_current
     assert c.read_noise == read_noise
     assert c.pixel_scale == pixel_scale
+    assert c.max_data_value == max_val
 
 
 def test_camera_unitscheck():
+    # Check that the units are checked properly
     gain = 2.0 * u.electron / u.adu
     read_noise = 10 * u.electron
     dark_current = 0.01 * u.electron / u.second
     pixel_scale = 0.563 * u.arcsec / u.pix
-
-    with pytest.raises(ValidationError, match="gain"):
-        c = Camera(
+    max_adu = 50000 * u.adu
+
+    # All 5 of the attributes after data_unit will be checked for units
```

Comment:
Cool, I didn't know you could do this with pytest, counting the number of errors.

Reply:
It isn't counting the errors, `pydantic` is -- `pytest` is just checking that the text of the error message contains `"5 validation errors"`